integration/kube_test.go

```diff
@@ -0,0 +1,132 @@
+
+	kube_api "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/golang/glog"
+	influxdb "github.com/influxdb/influxdb/client"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	influxdbController          = "monitoring-influxGrafanaController"
+	heapsterController          = "monitoring-heapsterController"
+	influxdbLabels              = map[string]string{"name": "influxGrafana"}
+	heapsterLabels              = map[string]string{"name": "heapster"}
+	kubeVersions                = flag.String("kube_versions", "0.7.2", "Comma separated list of kube versions to test against")
+	heapsterManifestFile        = flag.String("heapster_controller", "../deploy/heapster-controller.js", "Path to heapster replication controller file.")
+	influxdbGrafanaManifestFile = flag.String("influxdb_grafana_controller", "../deploy/influxdb-grafana-controller.js", "Path to Influxdb-Grafana replication controller file.")
+	influxdbServiceFile         = flag.String("influxdb_service", "../deploy/influxdb-service.json", "Path to Inlufxdb service file.")
+)
+
+func waitUntilPodRunning(fm kubeFramework, podLabels map[string]string, timeout time.Duration) (string, error) {
+	podsInterface := fm.Client().Pods(kube_api.NamespaceDefault)
+	for i := 0; i < int(timeout/time.Second); i++ {
+		selector := labels.Set(podLabels).AsSelector()
+		podList, err := podsInterface.List(selector)
+		if err != nil {
+			glog.V(1).Info(err)
+			return "", err
+		}
+		if len(podList.Items) != 1 {
+			glog.V(1).Info(err)
+			return "", fmt.Errorf("found %d pod with labels %v", len(podList.Items), podLabels)
+		}
+		podSpec := podList.Items[0]
+		glog.V(2).Infof("%+v", podSpec)
+		if podSpec.Status.Phase == kube_api.PodRunning {
+			return podSpec.Status.HostIP, nil
+		}
+		time.Sleep(time.Second)
+	}
+	return "", fmt.Errorf("pod not in running state after %d seconds", timeout/time.Second)
+}
+
+func updateReplicas(fm kubeFramework, name string, count int) error {
+	rcInterface := fm.Client().ReplicationControllers(kube_api.NamespaceDefault)
+	controller, err := rcInterface.Get(name)
+	if err == nil {
+		controller.Spec.Replicas = 0
+		if _, rerr := rcInterface.Update(controller); rerr != nil {
+			return rerr
+		}
+	}
+	glog.Errorf("updateReplicas failed for controller %+v with error: %q", controller, err)
+	return err
+}
+
+func deletePods(fm kubeFramework) {
+	// TODO(vishh): Fix this.
+	if err := updateReplicas(fm, influxdbController, 0); err != nil {
+		glog.Errorf("failed to bring down the number of replicas for influxdb")
+	}
+	if err := updateReplicas(fm, heapsterController, 0); err != nil {
+		glog.Errorf("failed to bring down the number of replicas for heapster")
+	}
+	_, _ = fm.RunKubectlCmd("delete", "-f", *heapsterManifestFile)
+	_, _ = fm.RunKubectlCmd("delete", "-f", *influxdbGrafanaManifestFile)
+	_, _ = fm.RunKubectlCmd("delete", "-f", *influxdbServiceFile)
+}
+
+func TestHeapsterInfluxDBWorks(t *testing.T) {
```

Comment:
consider doing:

require = require.New(t)

which lets you no longer have to specify t in the function